sds_data_manager/lambda_code/batch_starter.py

```diff
@@ -1,433 +1,364 @@
+    Appends start_time, end_time and version information to downstream dependents.
 
     Parameters
     ----------
-    cur : psycopg.extensions.cursor
-        A psycopg database cursor object to execute
-        database operations.
-    instrument : str
-        The name of the instrument for which details
-        are to be retrieved.
-    filename : str
-        The filename associated with the instrument.
-    process_range : int
-        Numbers of days backwards to process
-        e.g. 1 means process all data from 1 day ago
+    session : orm session
+        Database session.
+    downstream_dependents : list of dict
+        A list of dictionaries where each dictionary corresponds to a record
+    start_date : str
+        Start date of the event data.
+    end_date : str
+        End date of the event data.
+    version : str
+        Version of the event data.
 
     Returns
     -------
-    level : str
-        Instrument level
-    version : int
-        Version
-    process_dates : list
-        Dates to process
-    """
-
-    query = f"""SELECT * FROM sdc.{instrument.lower()}
-                WHERE filename = %s;"""
-    params = (filename,)
-
-    cur.execute(query, params)
-    column_names = [desc[0] for desc in cur.description]
-    records = cur.fetchall()
-
-    if not records:
-        raise ValueError(f"No records found for filename: {filename}")
-
-    # Check if more than one record is found
-    if len(records) > 1:
-        raise ValueError(
-            f"Expected a single record for filename: {filename}, "
-            f"but found multiple records."
-        )
-
-    record_dict = dict(zip(column_names, records[0]))
+    downstream_dependents : list of dict
+        Dictionary containing components with dates and versions appended.
 
-    level = record_dict["level"]
-    version = record_dict["version"]
+    """
 
-    date = record_dict["date"]
-    date_start = date - timedelta(days=process_range)
+    for dependent in downstream_dependents:
+        # TODO: query the version table here for appropriate version
+        #  of each downstream_dependent.
 
-    # Generate all the dates between date_start and date, inclusive
-    current_date = date_start
-    process_dates = []
+        # TODO: add repointing table query if dependent is ENA or GLOWS
+        #  Use start_date and end_date to query repointing table.
+        # Use pointing start_time and end_time in place of start_date and end_date.
+        # Add pointing number to dependent.
 
-    while current_date <= date:
-        process_dates.append(current_date.strftime("%Y-%m-%d"))
-        current_date += timedelta(days=1)
+        dependent["version"] = "v00-01"  # placeholder
+        dependent["start_date"] = start_date
+        dependent["end_date"] = end_date
 
-    return level, version, process_dates
+    return downstream_dependents
 
 
-def query_instruments(cur, version, process_dates, instruments):
+def find_dependencies(
```

Comment:
logically, I follow it. Later, we can simplify this search by making query to pre-processing table. We can modify this later post demo.